packages/kbn-monaco/src/types.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { monaco } from './monaco_imports';
+
+export interface LangModule {
+  ID: string;
+  lexerRules: monaco.languages.IMonarchLanguage;
+  languageConfiguration?: monaco.languages.LanguageConfiguration;
+  getSuggestionProvider?: Function;
+  getSyntaxErrors?: Function;
+}
+
+export interface CompleteLangModule extends LangModule {
```

Comment:
Why is this type needed? I assume it fixes a TS error related to the PainlessLang, since that's the only place it's being used?

Reply:
First I've started fixing those errors, but then I've recognized, that from the engineering perspective, in the future it can appear more complex configurations for languages, you'll want to support (as we have one already).

Reply:
`LangModuleType` will be used for simple or predefined  languages, which already exists in mocano.
`CompleteLangModuleType` will be used for complex, self-written languages.